app/models/apartment.rb

```diff
@@ -5,7 +5,28 @@ def get_reviews
     reviews = Review.where(:apartment_id => self.id)
     return reviews
   end
+  
   def review_count
     return Review.where(:apartment_id => self.id).size
   end
+  
+  def average_rating
+    avgRating = 0;
+    reviewCount = 0;
+    reviews = Review.where(:apartment_id => self.id)
+    reviews.each do |review|
```

Comment:
Is this nescessary? Can't you use SQL query to return the information you are looking for?

Reply:
This might be partially my fault since that is how I described it in the tasks.  But I do agree if this can be done with SQL it should be.

Reply:
I think this would be more efficient, but if it works it works.  

Additionally, I believe that the reading we had to do for reading 5 said that the fewest number of database accesses / queries would run the fastest. Depending on the differences in the computational steps required, this approach may be faster.  However as long as everything works I don't have any other issues than this one.

Reply:
I think there is a SQL query that can sum and I know there is one that can count, so you would only use two.  There might even be a built in sum function to ruby so you can do it in fewer lines of code with only one database access.

Reply:
I modeled my approach after some other syntax I saw in the code being used for similar things. Tested locally, this approach does work.

Reply:
Did a quick Google search and found this https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html .

It might be useful.